src/solver_nonlin_nox/4C_solver_nonlin_nox_vector.hpp

```diff
@@ -19,36 +22,99 @@ namespace NOX
 {
   namespace Nln
   {
-    class Vector : public ::NOX::Epetra::Vector
+    class Vector : public ::NOX::Abstract::Vector
     {
      public:
       enum class MemoryType
       {
-        View = ::NOX::Epetra::Vector::CreateView,
-        Copy = ::NOX::Epetra::Vector::CreateCopy
+        View,
+        Copy
       };
 
-      //! Reproduce ctors of ::NOX::Epetra::Vector
-      Vector(const Teuchos::RCP<Epetra_Vector>& source, MemoryType memoryType = MemoryType::Copy,
-          ::NOX::CopyType type = ::NOX::DeepCopy);
+      //! Ctor that get a shared_ptr with the state vector
+      Vector(const std::shared_ptr<Core::LinAlg::Vector<double>>& source,
+          MemoryType memory = MemoryType::Copy, ::NOX::CopyType type = ::NOX::DeepCopy);
+
+      //! Ctor that captures the ownership of the vector
+      Vector(Core::LinAlg::Vector<double>&& source);
 
-      Vector(const Epetra_Vector& source, ::NOX::CopyType type = ::NOX::DeepCopy);
+      //! Ctor that copies that provided state vector
+      Vector(const Core::LinAlg::Vector<double>& source, ::NOX::CopyType type = ::NOX::DeepCopy);
 
+      //! Ctor that copies from another NOX::Nln::Vector
       Vector(const NOX::Nln::Vector& source, ::NOX::CopyType type = ::NOX::DeepCopy);
 
       //! Main copy assignment operator
       ::NOX::Abstract::Vector& operator=(const NOX::Nln::Vector& source);
 
       //! Overloaded copy assignment operators
-      ::NOX::Abstract::Vector& operator=(const Epetra_Vector& source) override;
+      ::NOX::Abstract::Vector& operator=(const Core::LinAlg::Vector<double>& source);
       ::NOX::Abstract::Vector& operator=(const ::NOX::Abstract::Vector& source) override;
 
-      //! This operator is inherited and should not be ever called
-      ::NOX::Abstract::Vector& operator=(const ::NOX::Epetra::Vector&) override;
-
       //! Clone method
       Teuchos::RCP<::NOX::Abstract::Vector> clone(
           ::NOX::CopyType type = ::NOX::DeepCopy) const override;
+
+      //! Get reference to underlying Epetra vector.
+      Core::LinAlg::Vector<double>& get_linalg_vector();
+
+      //! Get const reference to underlying Epetra vector.
+      const Core::LinAlg::Vector<double>& get_linalg_vector() const;
+
+      //! Initialize every element of this vector with @p gamma .
+      ::NOX::Abstract::Vector& init(double gamma) override;
+
+      //! Initialize each element of this vector with a random value.
+      ::NOX::Abstract::Vector& random(bool, int) override;
```

Comment:
That's a pretty random function to have in a vector interface ;) What do I pass for `bool` and `int` here?